Kentico.Kontent.Management/Models/LanguageVariants/Elements/BaseElement.cs

```diff
@@ -20,6 +23,136 @@ public abstract class BaseElement
     /// </summary>
     public abstract dynamic ToDynamic();
 
+    /// <summary>
+    /// Transforms the dynamic object to the element.
+    /// </summary>
+    public static BaseElement FromDynamic(dynamic source, ElementMetadataType elementType) => elementType switch
+    {
+        ElementMetadataType.Text => FromDynamic(source, typeof(TextElement)),
+        ElementMetadataType.RichText => FromDynamic(source, typeof(RichTextElement)),
+        ElementMetadataType.Number => FromDynamic(source, typeof(NumberElement)),
+        ElementMetadataType.MultipleChoice => FromDynamic(source, typeof(MultipleChoiceElement)),
+        ElementMetadataType.DateTime => FromDynamic(source, typeof(DateTimeElement)),
+        ElementMetadataType.Asset => FromDynamic(source, typeof(AssetElement)),
+        ElementMetadataType.LinkedItems => FromDynamic(source, typeof(LinkedItemsElement)),
+        ElementMetadataType.Taxonomy => FromDynamic(source, typeof(TaxonomyElement)),
+        ElementMetadataType.UrlSlug => FromDynamic(source, typeof(UrlSlugElement)),
+        ElementMetadataType.Custom => FromDynamic(source, typeof(CustomElement)),
+        ElementMetadataType.Subpages => FromDynamic(source, typeof(SubpagesElement)),
+        _ => throw new InvalidOperationException("Unknown element type"),
+    };
+
+    /// <summary>
+    /// Transforms the dynamic object to the element.
+    /// </summary>
+    public static BaseElement FromDynamic(dynamic source, Type type)
+    {
+        if (type == typeof(TextElement))
+        {
+            return new TextElement
+            {
+                // TODO extend by codename + external ID
```

Comment:
After consulting with Ondra, the id does not need to be set, in that case the program should atlest throw some exception, or the other option is to add another option (e.g byDynamic) which finds out which of the reference variable is set and then use it.

Reply:
This is probably copy-pasted from ElementModelProvider, there, this comment makes sense as API always returns only Id. Now as we make it public to end-users the source might contain anything. We need to handle validation more robustly (e.g. what if the source object has a totally different shape than the type expects?). 


When it comes to the comment one possible approach is to try to find Id then Codename and lastly ExternalId and use the first to be found.

@winklertomas could you please have a look at the possible validation of inputs?

PS: The solution "it is good enough right now, can be improved later" is also valid.